build.sbt

```diff
@@ -111,35 +135,62 @@ lazy val testJS = test.js
 
 lazy val compilationSettings = Seq(
     maxErrors := 10
-  , scalacOptions in Compile ++= Seq(
-      "-deprecation"
-    , "-unchecked"
-    , "-feature"
-    , "-language:_"
-    , "-Ywarn-value-discard"
-    , "-Xlint"
-    , "-Xfatal-warnings"
-    ) ++ (
-      CrossVersion.partialVersion(scalaVersion.value) match {
-        case Some((2, 10)) =>
-          Seq("-Yno-adapted-args")
-        case Some((2, 13)) =>
-          Seq.empty
-        case _ =>
-          Seq("-Yno-adapted-args", "-Ywarn-unused-import")
-      }
-    )
+  , scalacOptions in Compile ++= (if (isDotty.value) {
+      Seq(
+        "-deprecation"
+        , "-unchecked"
+        , "-feature"
+        , "-Xfatal-warnings"
+        , "-source:3.0-migration"
+        , "-Ykind-projector"
+        , "-language:" + List(
+          "dynamics",
+          "existentials",
+          "higherKinds",
+          "reflectiveCalls",
+          "experimental.macros",
+          "implicitConversions"
+        ).mkString(",")
+        , "-siteroot", "./dotty-docs"
+      )
+    } else {
+      Seq(
+        "-deprecation"
+        , "-unchecked"
+        , "-feature"
+        , "-language:_"
+        , "-Ywarn-value-discard"
+        , "-Xlint"
+        , "-Xfatal-warnings"
+      ) ++ (
+        CrossVersion.partialVersion(scalaVersion.value) match {
+          case Some((2, 10)) =>
+            Seq("-Yno-adapted-args")
+          case Some((2, 13)) =>
+            Seq.empty
+          case _ =>
+            Seq("-Yno-adapted-args", "-Ywarn-unused-import")
+        }
+      )
+    })
   , scalacOptions in (Compile,console) := Seq("-language:_", "-feature")
   , scalacOptions in (Test,console) := Seq("-language:_", "-feature")
   , unmanagedSourceDirectories in Compile ++= {
       (unmanagedSourceDirectories in Compile).value.map { dir =>
         CrossVersion.partialVersion(scalaVersion.value) match {
-          case Some((2, 13)) => file(dir.getPath ++ "-2.13+")
+          case Some((2, 13)) | Some((3, 0)) => file(dir.getPath ++ "-2.13+")
```

Comment:
Should we match just on 3.x here to make it future compatible? Or not bother?

Reply:
As far as I know Scala 2.13 source code is compatible with Scala 3.0 so it should be fine. If it's just `scala-3.0`, I think, sbt can handle it.